tdns/ops_key.go

```
@@ -248,22 +249,22 @@ func (zd *ZoneData) BootstrapSig0KeyWithParent(alg uint8) (string, error, Update
 		}
 		zd.Logger.Printf(resp.Msg)
 	}
 
 	dump.P(ur)
-	return fmt.Sprintf("BootstrapSig0KeyWithParent(%s) sent update message; received rcode %s back", zd.ZoneName, dns.RcodeToString[rcode]), nil, ur
+	return fmt.Sprintf("BootstrapSig0KeyWithParent(%s) sent update message; received rcode %s back", zd.ZoneName, dns.RcodeToString[rcode]), ur, nil
 }
 
 // Returns msg, old keyid, new keyid, error, UpdateResult
-func (zd *ZoneData) RolloverSig0KeyWithParent(alg uint8, action string, oldkeyid, newkeyid uint16) (string, uint16, uint16, error, UpdateResult) {
+func (zd *ZoneData) RolloverSig0KeyWithParent(alg uint8, action string, oldkeyid, newkeyid uint16) (string, uint16, uint16, UpdateResult, error) {
```

golangci-lint (1.62.2)

SA4009: argument oldkeyid is overwritten before first use

(staticcheck)